documentation/templates/privilege_escalation.md.erb

```diff
@@ -99,3 +99,65 @@ This enables using any executable and command-line options to change users. Keep
 - Bolt always appends the specified `run-as` user to the specified `run-as-command`,
   which can be limiting for some escalating commands.
 - Bolt will not use the `run-as-command` option if `run-as` is not set.
+
+## Escalating privilege over WinRM
+
+The WinRM transport does not support `run-as`. This is because WinRM operates
+with a session level token and has no functionality to change to another user
+after authentication. Once you connect with a user, you can only perform
+actions that you have permission to perform with that user.  Instead of
+switching users, you can connect with a user, and then provide credentials to a
+program that accepts them to perform another action. The behavior of providing
+credentials to a program is not part of WinRM.
+
+For example, to get the processes running on a Windows target, you could
+connect to the target with WinRM using `Invoke-BoltScript -Script
+.\myscript.ps1 -Targets winrm://mytarget -User userA -Password passwordA` and
+then use the following Powershell script to pass credentials to the
+`Get-WmiObject` command:
+```
+$Cred = New-Object System.Management.Automation.PsCredential('userB',$passwordB)
+Get-WmiObject -Class Win32_BIOS -Computer SERVER1 -Credential $Cred
+```
+
+If you want to securely pass your userB credentials into a powershell script you have a few options.
+The best way to pass parameters securely into a script is by [converting it to a
+task](writing_tasks.md#converting-scripts-to-tasks) and [defining sensitive
+parameters](writing_tasks.md#defining-sensitive-parameters) for that task. This will obfuscate the
+secret in logs and API responses.
+
+You can pass sensitive parameters to a task directly if your secrets are the same for every target.
+If your secrets are unique to each target, or need to be computed, fetched, or obfuscated when you
+pass them to the task you will want to set your secret as data on the target in the inventory. For
+the purposes of escalating as a secondary user you can set arbitrary variables on targets in the
+inventory, which you can then access.
+
+There are a few ways of setting and retrieving variables on targets. You can set variables on a
+target using an [inventory plugin](using_plugins.md), or using [Hiera
+data](applying_manifest_blocks.md#using-hiera-data-in-a-manifest-block). You can then access those
+variables in a plan, or if you're using inventory plugins you can access them directly in the task
+using a [remote task](writing_tasks.md#writing-remote-tasks). Remote tasks run on localhost and pass
+the entire target object as json to the task as a metaparameter `_target`. You would write a task to
+get the secret, then connect to the remote target and use the secret there. Put another way, you
+have to manage the connection to the remote target in your task rather than letting Bolt manage it
+for you.
+
+To summarize:
```

Comment:
Instead of summarizing at the end, let's move the summary bit to the top - that's a good way to signpost to the reader that you're going to discuss two possible options/use cases